### PR TITLE
feat: Implement location feature in f-droid build

### LINKED ISSUE #2006: Add current location feature for f froid build
**Would you like to work on the issue?**

yes

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #2006 
 - Remove flavor specific module as not required.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
> Remove flavor specific module as not required.

Why not required?

**Comment by liveHarshit** [comment]
> > Remove flavor specific module as not required.
> 
> Why not required?

There was only one module which is configured for location feature, now the location feature is implemented in both variant in the same way, so I have removed the blank module

**Comment by iamareebjamal** [comment]
How was it implemented previously in Google Play Variant?

**Comment by liveHarshit** [comment]
> How was it implemented previously in Google Play Variant?

There was a ViewModel: WelcomeViewMode same copy of GeoLocationViewModel in flavor specific view model. Now I use the GeoLocationViewModel in place of that.


> now the location feature is implemented in both variant in the same way

The same way means ViewModel type is same but location service implementation is different. No change in location service implementation for play variant, I have just implemented the location service class for fdroid variant.

**Comment by liveHarshit** [comment]
Also, there were two welcome fragments for different variant, which is really not required now.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -7,6 +7,8 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
```
Comment (iamareebjamal) [inline app/src/main/AndroidManifest.xml]: No use of fine AFAIK, or is there?

Comment (liveHarshit) [reply app/src/main/AndroidManifest.xml]: Here `LocationManager.NETWORK_PROVIDER` is used so fine location is required.

Comment (liveHarshit) [reply app/src/main/AndroidManifest.xml]: According to https://developer.android.com/guide/topics/location/strategies.html

**Pull request merged**